subprojects/performance/src/performanceTest/groovy/org/gradle/performance/regression/inception/BuildSrcApiChangePerformanceTest.groovy

```diff
@@ -83,6 +83,12 @@ class BuildSrcApiChangePerformanceTest extends AbstractCrossVersionPerformanceTe
         }
     }
 
+    private static void useG1GarbageCollectorOnJava8(CrossVersionPerformanceTestRunner runner) {
+        if (!JavaVersion.current().isJava9Compatible()) {
+            runner.gradleOpts.addAll(['-XX:+UnlockExperimentalVMOptions', '-XX:+UseG1GC'])
```

Comment:
I think we really should avoiding to fine-tune GC parameters for our performance tests. If Gradle is generating too much garbage for the JVM to handle, then this is a real problem which we probably should fix.

Note that Android studio will also enable parallel GC for Java 11 when building from Android Studio, since they measured that this is faster. So that would mean that this test is not testing what our users will see.

My suggestion is to find out why we are creating more Garbage than parallel GC can handle and then fix that instead.

Reply:
I agree, we should look at what kind of objects Gradle creates in this scenario. At the same time, I don't think we'll find anything special here. This is a synthetic use-case where we push Gradle to the limit. We start multiple builds one after the other with changes enforcing buildSrc invalidation in a memory-constrained setup. This does not replicate any real-world scenario. And even if it happens somewhere out there the impact will be low: either the build will fail with a clear error message (please assign more memory to Gradle) or a new Daemon will spawn every now and then.
The aim of this PR to stabilize the the performance test results and no more.

Reply:
As long as there is a follow up investigation why we put the JVM under so much memory pressure we can put this fix into place. Though I think it should be a temporary fix.